dashboard/config/routes.rb

```diff
@@ -317,6 +317,13 @@ def redirect_to_teacher_dashboard
 
   post '/plc/course_units/:id/submit_new_questions_and_answers', to: 'plc/course_units#submit_new_questions_and_answers'
 
+  namespace :pd do
+    get 'workshops/:workshop_id/enroll', action: 'new', controller: 'workshop_enrollment'
+    post 'workshops/:workshop_id/enroll', action: 'create', controller: 'workshop_enrollment'
+    get 'workshop_enrollment/:code', action: 'show', controller: 'workshop_enrollment'
```

Comment:
I'm not too concerned because the worst you can do if you discover someone's enrollment code is to cancel it. In that case, they will get a cancellation email (once that part is complete, outside this PR). Also enrollments are not necessary for attendance. No personal information is displayed. 

For extra security we could get rid of the 404 and just act like the enrollment was cancelled even if the code is invalid. Thoughts?